### PR TITLE
list for GS

### PR DESCRIPTION
**Pull request by Guseyn**
list implementation for #119

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#122](https://github.com/artipie/asto/issues/122) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8244de32-50bd-4afc-a4b5-464a1ad62029); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/asto/google/GoogleStorage.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        this.bucket = bucket;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Collection<Key>> list(final Key prefix) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/asto/google/GoogleStorage.java]: @Guseyn can we add a test for this method?

**Comment by Vatavuk** [comment]
@Guseyn you have merge confilcts

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by Vatavuk** [comment]
@0crat wait

**Comment by 0crat** [comment]
> @0crat wait ([here](https://github.com/artipie/asto/issues/122#issuecomment-605634160))

@Vatavuk It's a code review job [#122](https://github.com/artipie/asto/issues/122), can't put it on hold

**Comment by Vatavuk** [comment]
@Guseyn what about this PR? Please update it. I'm ok if you add a test in a separate PR.

**Comment by Vatavuk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/asto/issues/122#issuecomment-606673578))

@Vatavuk The user @vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#122](https://github.com/artipie/asto/issues/122), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#122](https://github.com/artipie/asto/issues/122) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8473e65f-9f13-4c0e-983d-548588726f60); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by victornoel** [comment]
@Guseyn as a started, could you fix the conflict and add `@todo` so that the work can be continued?

@g4s8 will you merge this PR since the corresponding ticket is not in scope?

**Comment by victornoel** [comment]
@g4s8 I think this PR should be refused, author doe not answer and it's not in scope.

**Comment by g4s8** [comment]
@Guseyn @victornoel closing as stale PR

**Comment by 0crat** [comment]
Job was finished in 46 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Pull request [#122](https://github.com/artipie/asto/issues/122) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by sereshqua** [comment]
@Guseyn please make sure you will fix all problems found by code reviewers next time

**Comment by victornoel** [comment]
@sereshqua I don't think he is going to answer now, can we go on with QA? :)

**Comment by sereshqua** [comment]
@victornoel let's wait at least 3 days, cause policy recommends NOT to give verdicts straightaway :)

**Comment by sereshqua** [comment]
@Guseyn ping

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request closed without merging**